core/controllers/reader.py

```diff
@@ -310,68 +424,58 @@ def post(self, exploration_id):
+            playthrough = stats_domain.Playthrough.from_dict(playthrough_data)
+            orig_playthrough = stats_services.get_playthrough_by_id(
+                playthrough_id)
+            if orig_playthrough.issue_type != playthrough.issue_type:
+                self._move_playthrough_to_correct_issue(
+                    playthrough, orig_playthrough)
+
+            stats_services.update_playthroughs_multi(
+                [playthrough_data['id']], [playthrough])
+            stats_services.save_exp_issues_model_transactional(self.exp_issues)
+            self.render_json({})
+            return
 
-        customization_args = exp_issue_dict['issue_customization_args']
+        playthrough_id = None
+        payload_return = {'playthrough_stored': True}
 
-        issue_found = False
-        for index, issue in enumerate(exp_issues.unresolved_issues):
-            if issue.issue_type == exp_issue_dict['issue_type']:
-                issue_customization_args = issue.issue_customization_args
-                # In case issue_keyname is 'state_names', the ordering of the
-                # list is important i.e. [a,b,c] is different from [b,c,a].
-                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[
-                    issue.issue_type]
-                if (issue_customization_args[issue_keyname] ==
-                        customization_args[issue_keyname]):
-                    issue_found = True
-                    if (len(issue.playthrough_ids) <
-                            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
-                        playthrough_id = stats_models.PlaythroughModel.create(
-                            playthrough_data['exp_id'],
-                            playthrough_data['exp_version'],
-                            playthrough_data['issue_type'],
-                            playthrough_data['issue_customization_args'],
-                            playthrough_data['actions'])
-                        exp_issues.unresolved_issues[
-                            index].playthrough_ids.append(playthrough_id)
-                    break
-
-        if not issue_found:
-            playthrough_id = stats_models.PlaythroughModel.create(
-                playthrough_data['exp_id'],
-                playthrough_data['exp_version'],
-                playthrough_data['issue_type'],
-                playthrough_data['issue_customization_args'],
-                playthrough_data['actions'])
-            issue = stats_domain.ExplorationIssue(
-                exp_issue_dict['issue_type'],
-                exp_issue_dict['issue_customization_args'],
-                [playthrough_id], exp_issue_dict['schema_version'],
-                exp_issue_dict['is_valid'])
-
-            exp_issues.unresolved_issues.append(issue)
+        # Create a dummy playthrough domain object with a temporary ID since the
+        # playthrough doesn't exist yet.
+        playthrough_data['id'] = stats_models.TEMPORARY_PLAYTHROUGH_ID
```

Comment:
BTW, what was the conclusion of @seanlip's suggestion? Can we avoid using a temp ID?

Reply:
@brianrodri I can't seem to think of an immediate work around here t least. We cant pass the dict explicitly either, because the sub function calls another function with the playthrough domain object as parameter. If we were to be passing around dicts so much, it wouldn't be a nice thing to do.

Reply:
I'm not sure I see the issue w/ passing dicts around, there's nothing wrong w/ that.

Reply:
Actually there is -- we prefer domain objects to dicts because they encapsulate behaviour (at least, for dicts that represent objects).

So here's the thing -- what is the definition of a Playthrough? In particular, does it include the ID or not?

If it does include the ID, then it's not a playthrough before you assign the ID. In which case you probably want to refactor self._assign_playthrough_to_issue to take individual args rather than a "fake playthrough" object.

If it does not include the ID, then that ID should be removed from the domain object.

So ... which is it?

Reply:
@seanlip 
The Playthrough domain object does require the ID. This instance is just one particular use case where we don't use an ID. The ID is required in a  lot of other occurrences where we use the playthrough domain object.

So, I guess I could go ahead and pass individual args?

But then, I wouldn't call this a fake playthrough instance? Cant we consider it as a temporary instance which would not be committeed to the datastore as is?

Reply:
Yes, pass individual args. This is not a fake playthrough instance or a temporary instance. By definition, it's not a playthrough at all.

Reply:
Resolved.